dashboard/src/components/skills/selfReport/SelfReportApproval.vue

```diff
@@ -0,0 +1,229 @@
+        <b-button variant="outline-info" @click="changeSelectionForAll(true)" data-cy="selectPageOfApprovalsBtn" class="mr-2 mt-1"><i class="fa fa-check-square"/> Select Page</b-button>
+        <b-button variant="outline-info" @click="changeSelectionForAll(false)" data-cy="clearSelectedApprovalsBtn" class="mt-1"><i class="far fa-square"></i> Clear</b-button>
+      </div>
+      <div class="col text-right">
+        <b-button variant="outline-danger" @click="reject.showModal=true" data-cy="rejectBtn" class="mt-1" :disabled="actionsDisabled"><i class="fa fa-times-circle"/> Reject</b-button>
+        <b-button variant="outline-success" @click="approve" data-cy="approveBtn" class="mt-1 ml-2" :disabled="actionsDisabled"><i class="fa fa-check"/> Approve</b-button>
+      </div>
+    </div>
+
+    <skills-b-table :options="table.options" :items="table.items"
+                    @page-changed="pageChanged"
+                    @page-size-changed="pageSizeChanged"
+                    @sort-changed="sortTable"
+                    data-cy="skillsReportApprovalTable">
+
+      <template v-slot:cell(userId)="data">
+        <b-form-checkbox
+          :id="`${data.item.userId}-${data.item.skillId}`"
+          v-model="data.item.selected"
+          :name="`checkbox--${data.item.skillId}`"
+          :value="true"
+          :unchecked-value="false"
+          :inline="true"
+          v-on:input="updateActionsDisableStatus"
+          :data-cy="`approvalSelect_${data.item.userId}-${data.item.skillId}`"
+        >
+          {{ data.value }}
+        </b-form-checkbox>
+      </template>
+
+      <template v-slot:cell(request)="data">
+        <div>{{ data.item.skillName }}</div>
+        <div class="small text-secondary">ID: {{ data.item.skillId }}</div>
+        <div class="mt-2" style="font-size: 0.9rem;"><span class="text-secondary">Note:</span>
+          <span v-if="data.item.requestMsg && data.item.requestMsg.length > 0"> {{ data.item.requestMsg }}</span>
+          <span v-else class="text-muted"> Not supplied</span>
+        </div>
+      </template>
+
+      <template v-slot:cell(requestedOn)="data">
+        <date-cell :value="data.value" />
+      </template>
+
+    </skills-b-table>
+
+    <ValidationObserver v-slot="{invalid}" slim>
+    <b-modal id="rejectSkillsModal"
+             title="REJECT SKILLS"
+             :no-close-on-backdrop="true"
+             v-model="reject.showModal">
+        <div class="row p-2" data-cy="rejectionTitle">
+          <div class="col-auto text-center">
+            <i class="far fa-thumbs-down text-warning" style="font-size: 3rem"/>
+          </div>
+          <div class="col">
+            <p class="h6">This will <b class="text-warning">permanently</b> reject user requests to get points. Users will be notified and you can provide an optional message below.</p>
```

Comment:
does this apply to a single request or all requests? If single, text should be changed to "This will permanently reject this user's request for points..."

Reply:
it applies to all the selected requests